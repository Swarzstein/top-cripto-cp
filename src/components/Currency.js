/* eslint-disable react/prop-types */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
// import { useParams } from 'react-router-dom';
import { showDetails } from '../redux/details/details';

export default function Currency(data) {
  const { currency, bg } = data;
  const {
    id, rank, name, symbol, price,
  } = currency;
  // console.log(currency);
  const dispatch = useDispatch();
  const currencyDetails = (currency) => {
    dispatch(showDetails(currency));
  };

  return (
    <Link to={`/${id}`}>
      <div role="presentation" className={`currency-box ${bg}`} onClick={() => currencyDetails(currency)} id={id}>
        <h4>
          #
          {rank}
        </h4>
        <h2>{symbol}</h2>
        <h3>{name}</h3>
        <p>{price}</p>
      </div>
    </Link>
  );
}

Currency.defaultProps = {
  rank: '',
  symbol: '',
  name: '',
  price: '',
};

Currency.prop = {
  rank: propTypes.number,
  symbol: propTypes.string,
  name: propTypes.string,
  price: propTypes.number,
};
