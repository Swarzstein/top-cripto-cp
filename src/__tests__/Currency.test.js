import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Currency from '../components/Currency';
import store from '../redux/configureStore';

describe('Currency component', () => {
  it('Currency box renders', () => {
    const currency = {
      id: 'bitcoin',
      rank: '1',
      symbol: 'BTC',
      name: 'Bitcoin',
      supply: '19288081.0000000000000000',
      maxSupply: '21000000.0000000000000000',
      marketCapUsd: '421510245908.9933298571526450',
      volumeUsd24Hr: '6220421918.5937454009328059',
      priceUsd: '21853.4050074236690450',
      changePercent24Hr: '0.9239425835453880',
      vwap24Hr: '21749.1227056822936267',
      explorer: 'https://blockchain.info/',
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Currency currency={currency} />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
