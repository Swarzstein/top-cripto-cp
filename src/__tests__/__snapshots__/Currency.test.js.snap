// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency component Currency box renders 1`] = `
<a
  href="/bitcoin"
  onClick={[Function]}
>
  <div
    className="currency-box undefined"
    id="bitcoin"
    role="presentation"
  >
    <h4>
      #
      1
    </h4>
    <h2>
      BTC
    </h2>
    <h3>
      Bitcoin
    </h3>
    <p>
      21853.4050074236690450
    </p>
  </div>
</a>
`;
